packages/ag-charts-enterprise/src/features/zoom/zoomPanner.ts

```diff
@@ -1,19 +1,136 @@
 
+const { RATIO, Validate } = _ModuleSupport;
+export interface ZoomPanUpdate {
+    type: 'update';
+    deltaX: number;
+    deltaY: number;
+}
+
+interface ZoomCoordHistory {
+    x: number;
+    y: number;
+    t: number;
+}
+
+const maxZoomCoords = 16;
+
 export class ZoomPanner {
+    @Validate(RATIO)
+    deceleration: number = 1;
+
+    private onUpdate: ((e: ZoomPanUpdate) => void) | undefined;
+
     private coords?: ZoomCoords;
 
-    update(event: _ModuleSupport.InteractionEvent<'drag'>, bbox: _Scene.BBox, zooms: AxisZoomStates): AxisZoomStates {
+    private coordsMonitorTimeout: NodeJS.Timeout | undefined;
+    private zoomCoordsHistoryIndex = 0;
+    private coordsHistory: ZoomCoordHistory[] = [];
+
+    private inertiaHandle: number | undefined;
+
+    addListener(_type: 'update', fn: (e: ZoomPanUpdate) => void) {
+        this.onUpdate = fn;
+        return () => {
+            this.onUpdate = undefined;
+        };
+    }
+
+    stopInteractions() {
+        if (this.inertiaHandle != null) {
+            cancelAnimationFrame(this.inertiaHandle);
+            this.inertiaHandle = undefined;
+        }
+    }
+
+    update(event: _ModuleSupport.InteractionEvent<'drag'>) {
         this.updateCoords(event.offsetX, event.offsetY);
-        return this.translateZooms(bbox, zooms);
+        const { x1 = 0, y1 = 0, x2 = 0, y2 = 0 } = this.coords ?? {};
+        this.onUpdate?.({
+            type: 'update',
+            deltaX: x1 - x2,
+            deltaY: y1 - y2,
+        });
+    }
+
+    start() {
+        this.coordsMonitorTimeout = setInterval(this.recordCurrentZoomCoords.bind(this), 16);
     }
 
     stop() {
+        const { coordsHistory } = this;
+
+        let deltaX = 0;
+        let deltaY = 0;
+        let deltaT = 0;
+        if (coordsHistory.length > 0) {
+            const arrayIndex = this.zoomCoordsHistoryIndex % maxZoomCoords;
+            let index1 = arrayIndex - 1;
+            if (index1 < 0) index1 = coordsHistory.length - 1;
+            let index0 = arrayIndex;
+            if (index0 >= coordsHistory.length) index0 = 0;
+
+            const coords1 = coordsHistory[index1];
+            const coords0 = coordsHistory[index0];
+
+            deltaX = coords1.x - coords0.x;
+            deltaY = coords1.y - coords0.y;
+            deltaT = coords1.t - coords0.t;
+        }
+
         this.coords = undefined;
+        clearInterval(this.coordsMonitorTimeout);
+        this.coordsMonitorTimeout = undefined;
+        this.zoomCoordsHistoryIndex = 0;
+        this.coordsHistory.length = 0;
+
+        if (deltaT > 0 && this.deceleration < 1) {
+            const xVelocity = deltaX / deltaT;
+            const yVelocity = deltaY / deltaT;
+            const velocity = Math.hypot(xVelocity, yVelocity);
+            const angle = Math.atan2(yVelocity, xVelocity);
+            const t0 = performance.now();
+            this.inertiaHandle = requestAnimationFrame((t) => {
```

Comment:
I'm not sure if we can use `requestAnimationFrame()` directly (here and below) - if chart rendering is slower than 60 FPS, I'm concerned that the updates this generates will not play well with an in-progress Chart.performUpdate() cycle.

Could we adjust the solution to use `AnimationManager` to execute the animation steps, as it already handles coordination with `Chart.performUpdate()`?

Reply:
Discussed offline - as this uses `ZoomManager` to percolate state changes, we are already taking this concern indirectly into account.